TeamCode/src/main/java/org/firstinspires/ftc/teamcode/auton/vision/AprilTagAlignment.java

```diff
@@ -0,0 +1,278 @@
+        if(xDistance < targetBearing) {
+            leftFTarget = xDistance;
+            rightFTarget = -xDistance;
+            leftBTarget = -xDistance;
+            rightBTarget = xDistance;
+        } else { // Move robot right.
+            leftFTarget = -xDistance;
+            rightFTarget = xDistance;
+            leftBTarget = xDistance;
+            rightBTarget = -xDistance;
+        }
+
+        setTargetPosition(robot.<DcMotor>get("leftFront"), leftFTarget);
+        setTargetPosition(robot.<DcMotor>get("rightFront"), rightFTarget);
+        setTargetPosition(robot.<DcMotor>get("leftRear"), leftBTarget);
+        setTargetPosition(robot.<DcMotor>get("rightRear"), rightBTarget);
+
+        while (robot.<DcMotor>get("leftFront").getTargetPosition() !=
+                robot.<DcMotor>get("leftFront").getCurrentPosition() &&
+                robot.<DcMotor>get("rightFront").getTargetPosition() !=
+                    robot.<DcMotor>get("rightFront").getCurrentPosition() &&
+                robot.<DcMotor>get("leftRear").getTargetPosition() !=
+                    robot.<DcMotor>get("leftRear").getCurrentPosition() &&
+                robot.<DcMotor>get("rightRear").getTargetPosition() !=
+                    robot.<DcMotor>get("rightRear").getCurrentPosition()) {
+
+            showMotorStatus(robot.<DcMotor>get("leftFront")); // Also turns on/off motor power.
+            showMotorStatus(robot.<DcMotor>get("rightFront"));
+            showMotorStatus(robot.<DcMotor>get("leftRear"));
+            showMotorStatus(robot.<DcMotor>get("rightRear"));
+        }
+    }
+
+    public void moveRange() {
+        double distance = inchesToMM(targetTag.ftcPose.range);
+
+        setTargetPosition(robot.<DcMotor>get("leftFront"), distance);
+        setTargetPosition(robot.<DcMotor>get("rightFront"), distance);
+        setTargetPosition(robot.<DcMotor>get("leftRear"), distance);
+        setTargetPosition(robot.<DcMotor>get("rightRear"), distance);
+
+        while (robot.<DcMotor>get("leftFront").getTargetPosition() !=
+                robot.<DcMotor>get("leftFront").getCurrentPosition() &&
+                robot.<DcMotor>get("rightFront").getTargetPosition() !=
+                        robot.<DcMotor>get("rightFront").getCurrentPosition() &&
+                robot.<DcMotor>get("leftRear").getTargetPosition() !=
+                        robot.<DcMotor>get("leftRear").getCurrentPosition() &&
+                robot.<DcMotor>get("rightRear").getTargetPosition() !=
+                        robot.<DcMotor>get("rightRear").getCurrentPosition()) {
+
+            showMotorStatus(robot.<DcMotor>get("leftFront")); // Also turns on/off motor power.
+            showMotorStatus(robot.<DcMotor>get("rightFront"));
+            showMotorStatus(robot.<DcMotor>get("leftRear"));
+            showMotorStatus(robot.<DcMotor>get("rightRear"));
+        }
+    }
+
+    public void showMotorStatus(DcMotor motor) {
```

Comment:
Why is `showMotorStatus` also changing the motor power? This seems iffy.

Reply:
yeah that's fair